web/client-api/src/main/java/io/deephaven/web/client/api/parse/JsDataHandler.java

```diff
@@ -136,12 +136,23 @@ private long parseDateString(String str, ParseContext context) {
                 dateTimeString = s.substring(0, spaceIndex);
                 timeZoneString = s.substring(spaceIndex + 1);
             }
-            final String pattern = getSubsecondPattern(dateTimeString, context);
+
             final TimeZone timeZone = timeZoneString == null
                     ? context.timeZone.unwrap()
                     : JsTimeZone.getTimeZone(timeZoneString).unwrap();
-            return JsDateTimeFormat.getFormat(pattern).parseWithTimezoneAsLong(dateTimeString, timeZone,
-                    JsTimeZone.needsDstAdjustment(timeZoneString));
+            boolean needsAdjustment = JsTimeZone.needsDstAdjustment(timeZoneString);
+
+            try {
+                // First try with the pattern we already have
+                return JsDateTimeFormat.getFormat(context.dateTimePattern).parseWithTimezoneAsLong(dateTimeString,
+                        timeZone, needsAdjustment);
+            } catch (IllegalArgumentException e) {
+                // We failed to parse with the existing context pattern, try and update the pattern from the string of
+                // text and do it again
+                context.dateTimePattern = getSubsecondPattern(dateTimeString);
```

Comment:
if the format was already something custom but missing subseconds, we're reverting to ISO and attempting to add subseconds to that?

Reply:
The `getSubsecondPattern` basically just determines what pattern is appropriate given the string, adding the amount of subseconds necessary